chia/wallet/util/clvm_streamable.py

```diff
@@ -1,145 +1,109 @@
+) -> Union[str, Dict[str, Any]]:
+    if next_recursion_step is None:
+        next_recursion_step = recurse_jsonify
+    if hasattr(streamable, "__clvm_streamable__"):
+        # If we are using clvm_serde, we stop JSON serialization at this point and instead return the clvm blob
+        return byte_serialize_clvm_streamable(streamable).hex()
+    else:
+        return next_recursion_step(streamable, json_serialize_with_clvm_streamable)
+
+
+def program_deserialize_clvm_streamable(program: Program, clvm_streamable_type: Type[_T_Streamable]) -> _T_Streamable:
+    # Underlying hinting problem with clvm_serde
+    return from_program_for_type(clvm_streamable_type)(program)  # type: ignore[no-any-return]
+
+
+def byte_deserialize_clvm_streamable(blob: bytes, clvm_streamable_type: Type[_T_Streamable]) -> _T_Streamable:
+    return program_deserialize_clvm_streamable(Program.from_bytes(blob), clvm_streamable_type)
+
+
+def is_compound_type(typ: Any) -> bool:
+    return is_type_SpecificOptional(typ) or is_type_Tuple(typ) or is_type_List(typ)
+
+
+def json_deserialize_with_clvm_streamable(
+    json_dict: Union[str, Dict[str, Any]], streamable_type: Type[_T_Streamable]
+) -> _T_Streamable:
+    if isinstance(json_dict, str):
+        return byte_deserialize_clvm_streamable(bytes.fromhex(json_dict), streamable_type)
+    else:
+
+        def bind_type_to_this_function(
+            type: Type[object],
+        ) -> Callable[[object], Streamable]:
+            def convert_function(item: object) -> Streamable:
+                # Type ignore due to underlying incompatibilites with streamable library
+                return json_deserialize_with_clvm_streamable(item, type)  # type: ignore
+
+            return convert_function
+
+        old_streamable_fields = streamable_type.streamable_fields()
+        new_streamable_fields = []
+        for old_field in old_streamable_fields:
+            if is_compound_type(old_field.type):
+                inner_type = get_args(old_field.type)[0]
+                if hasattr(inner_type, "__clvm_streamable__"):
+                    new_streamable_fields.append(
+                        dataclasses.replace(
+                            old_field,
+                            convert_function=function_to_convert_one_item(
+                                old_field.type, bind_type_to_this_function(inner_type)
+                            ),
+                        )
+                    )
+                else:
+                    new_streamable_fields.append(old_field)
+            elif hasattr(old_field.type, "__clvm_streamable__"):
+                new_streamable_fields.append(
+                    dataclasses.replace(old_field, convert_function=bind_type_to_this_function(old_field.type))
+                )
+            else:
+                new_streamable_fields.append(old_field)
+
+        setattr(streamable_type, "_streamable_fields", tuple(new_streamable_fields))
```

Comment:
Can we fill out all this information at the time of decoration?  As is, I think we are redoing all this analysis, including reprocessing the already processed streamable fields (?), every time we deserialize to this type.

Maybe streamable should provide an interface for modifying it's private data.  That way it doesn't have to worry about that attribute format being incompatible with other code.

Reply:
We cannot unfortunately.  If we did this at decoration time, then the `.to_json_dict` function would only serialize in the clvm_streamable way.  What we are doing here is updating all of the `_streamable_fields` on the fly to use this function for conversion rather than the default it uses to convert from json.

Reply:
Sure, streamable may need to be aware etc.  Or this layer may need to do some extra work.  I assume we'll end up going with this, but reprocessing the classes every time we need to serialize them was a thing we were very happy to get out of streamable.

Reply:
Yes, I can see that, I guess one of my objectives here is to make sure all of the baggage stays with plain old streamable and this doesn't add new baggage to the library.  I want as light of a touch on streamable as possible so that if it gets reworked, this can be treated simply as a downstream dependency rather than influencing the design of streamable at all.

Reply:
I'm not clear how this is just a downstream dependency given that it both reads _and writes_ internal streamable data.  But ok, feel free to resolve this conversation.

Reply:
The writing of the internal data is temporary to this function so it's just a confusing usage of a confusing API.